### PR TITLE
Configurable Consul Service Address

### PR DESCRIPTION
**Pull request by trotttrotttrott**
Setting an explicit service address makes Consul ignore the [`translate_wan_addrs`](https://www.consul.io/docs/agent/options.html#translate_wan_addrs) setting. When set, Consul nodes will return their WAN address on queries from foreign datacenters. This seems to me to be a nice option to have instead of defining the `redirectHost` as the service-specific address.

This behavior is mentioned here: https://github.com/hashicorp/consul/issues/2390#issuecomment-252077332.

### CODE REVIEW

**Comment by jefferai** [comment]
I'm very wary of this change as it may well break things for other people. I think the right way to go would be to instead add a `service_address` parameter to Consul configuration. If it's set, use that including if it's set but blank; otherwise fall back to using the redirect addr.

**Comment by trotttrotttrott** [comment]
@jefferai yeah, that's a good call. Thanks for the feedback. I'll see if I can get that worked out 👍

**Comment by trotttrotttrott** [comment]
@jefferai I think this is ready for another review.

Something to note is that the service ID will always use the convention, `<service_name>:<redirect_addr>:<redirect_port>`. This could be a little odd if `service_address` is actually set. I don't think it's a problem, but it's possibly not ideal.

File: physical/consul/consul.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ type ConsulBackend struct {
 	redirectPort        int64
 	serviceName         string
 	serviceTags         []string
+	serviceAddress      *string
```
Comment (jefferai) [inline physical/consul/consul.go]: This doesn't need to be a pointer and I'd prefer if it wasn't.

Comment (jefferai) [reply physical/consul/consul.go]: Actually nm; I remember now why this is a pointer.

File: physical/consul/consul.go [hunk 2]
```diff
@@ -150,11 +151,22 @@ func NewConsulBackend(conf map[string]string, logger log.Logger) (physical.Backe
...
 	if logger.IsDebug() {
 		logger.Debug("physical/consul: config service_tags set", "service_tags", tags)
 	}
 
+	// Get the service-specific address to override the use of the HA redirect address
+	var serviceAddr *string
+	serviceAddrStr, ok := conf["service_address"]
+	if ok {
+		serviceAddr = &serviceAddrStr
+	} else {
```
Comment (jefferai) [inline physical/consul/consul.go]: This else block is unnecessary as nil is the default value.

Comment (trotttrotttrott) [reply physical/consul/consul.go]: @jefferai thanks! Just removed the else block 👍

**Comment by jefferai** [comment]
Thanks!

**Pull request merged**